infrastructure/xai-relayer-api/src/loaders/mongoose.ts

```diff
@@ -1,11 +1,16 @@
-import mongoose, {ConnectOptions, Connection} from "mongoose";
+import mongoose, { ConnectOptions, Connection } from "mongoose";
 
 export async function disconnectMongoose(): Promise<void> {
 	await mongoose.disconnect();
 	console.log("Mongoose disconnected");
 }
 
+let cachedConnection: typeof mongoose;
+
 export async function loadMongoose(): Promise<void> {
+	if (cachedConnection) {
+		return Promise.resolve();
```

Comment:
not a big deal to refactor for this PR, but I don't think you need the Promise.resolve(), since this is an Async function, its already wrapped in a promise.